test/performance/range/gap_decorator_random_benchmark.cpp

```diff
@@ -0,0 +1,283 @@
+// 1 b) Read at random position in gapped sequence
+BENCHMARK_TEMPLATE(read_random, gap_decorator_anchor_set<const std::vector<dna4> &>, true)->Range(1<<2, 1<<15);
+BENCHMARK_TEMPLATE(read_random, std::vector<gapped<dna4>>, true)->Range(1<<2, 1<<15);
+
+// ============================================================================
+//  insert at random position
+// ============================================================================
+template <typename gap_decorator_t, bool gapped_flag>
+static void insert_random(benchmark::State& state)
+{
+    unsigned int seq_len = state.range(0);
+    using size_type = typename gap_decorator_t::size_type;
+    using sequence_type = remove_cvref_t<detail::unaligned_seq_t<gap_decorator_t>>;
+    sequence_type seq(seq_len, 'A'_dna4);
+
+    // vector of sampled gap lengths for each position
+    std::vector<size_type> gaps(seq_len, 0);
+    sample<size_type>(gaps, seq_len);
+
+    // determine sum of gaps and non-gap symbols for not exceeding targeted sequence length
+    if constexpr(gapped_flag)
+        resize<size_type, sequence_type>(gaps, seq, seq_len);
+
+    // initialize with (truncated) sequence and insert gaps from left to right
+    gap_decorator_t gap_decorator;
+    assign_unaligned(gap_decorator, seq);
+
+    // insert gaps before starting benchmark
+    if constexpr(gapped_flag)
+        insert_gaps<size_type, gap_decorator_t>(gaps, gap_decorator);
+
+    std::mt19937 generator(time(0)); //Standard mersenne_twister_engine seeded with current time
+    std::uniform_real_distribution<> uni_dis(0.0, static_cast<double>(seq_len));
+
+    size_t op_ctr = 0;
+    for (auto _ : state)
+    {
+        state.PauseTiming();
+        size_t pos = uni_dis(generator);
+        auto it = std::ranges::begin(gap_decorator);
+        std::ranges::advance(it, pos);
+        state.ResumeTiming();
+        insert_gap(gap_decorator, it, 1);
+        state.PauseTiming();
+        ++op_ctr;
+        state.ResumeTiming();
+    }
+    state.counters["insert_op"] = op_ctr;
+}
+
+// 2 a) Insert gaps of length 1 at random position into ungapped sequence
+BENCHMARK_TEMPLATE(insert_random, gap_decorator_anchor_set<const std::vector<dna4> &>, false)->Range(1<<2, 1<<15);
+BENCHMARK_TEMPLATE(insert_random, std::vector<gapped<dna4>>, false)->Range(1<<2, 1<<15);
+// 2 b) Insert gaps of length 1 at random position into gapped sequence
+BENCHMARK_TEMPLATE(insert_random, gap_decorator_anchor_set<const std::vector<dna4> &>, true)->Range(1<<2, 1<<15);
+BENCHMARK_TEMPLATE(insert_random, std::vector<gapped<dna4>>, true)->Range(1<<2, 1<<15);
+
+// ============================================================================
+//  delete at random position
+// ============================================================================
+template <typename gap_decorator_t, bool gapped_flag>
+static void delete_random(benchmark::State& state)
+{
+    unsigned int seq_len = state.range(0);
```

Comment:
The next 13 lines of code are exactly the same as in the other benchmarks, right? Can you also outsource them into a helper function to reduce code duplication?

In addition, (If I understand correctly) you only need to call `sample<size_type>(gaps, seq_len);` if you really want to insert gaps? You could then include this in the `if constexpr` branch which probably reduces (preparation) runtime in the ungapped case

```cpp
    if constexpr(gapped_flag)
    {
        // vector of sampled gap lengths for each position
        std::vector<size_type> gaps(seq_len, 0);
        sample<size_type>(gaps, seq_len);
        resize<size_type, sequence_type>(gaps, seq, seq_len);
    }
```